src/test/java/co/rsk/bitcoinj/script/FlyoverRedeemScriptParserTest.java

```diff
@@ -0,0 +1,226 @@
+        flyoverP2shErpRedeemScript = RedeemScriptUtils.createFlyoverRedeemScript(derivationArgumentsHash.getBytes(), p2shErpRedeemScript);
+
+        flyoverRedeemScripts = Arrays.asList(flyoverStandardRedeemScript, flyoverErpRedeemScript, flyoverP2shErpRedeemScript);
+
+        Script malformedScriptZeroSize = new Script(new byte[0]);
+        Script malformedScriptTwoSize = new Script(new byte[2]);
+
+        malformedScripts = Arrays.asList(malformedScriptZeroSize, malformedScriptTwoSize);
+    }
+
+    @Test
+    public void getMultiSigType_whenIsValidRedeemScript_shouldReturnFlyoverMultiSigType() {
+        for(Script flyoverRedeemScript : flyoverRedeemScripts) {
+            // Arrange
+            FlyoverRedeemScriptParser flyoverRedeemScriptParser = new FlyoverRedeemScriptParser(flyoverRedeemScript.getChunks());
+
+            // Act
+            MultiSigType actualMultiSigType = flyoverRedeemScriptParser.getMultiSigType();
+
+            // Assert
+            assertEquals(MultiSigType.FLYOVER, actualMultiSigType);
+        }
+    }
+
+    @Test
+    public void getM_whenFlyoverRedeemScriptContainsValidRedeemScript_shouldReturnMValue() {
+        // Arrange
+        final int EXPECTED_M = 5;
+        for (Script flyoverRedeemScript : flyoverRedeemScripts) {
+            FlyoverRedeemScriptParser flyoverRedeemScriptParser = new FlyoverRedeemScriptParser(flyoverRedeemScript.getChunks());
+
+            // Act
+            int actualM = flyoverRedeemScriptParser.getM();
+
+            // Assert
+            assertEquals(EXPECTED_M, actualM);
+        }
+    }
+
+    @Test
+    public void findKeyInRedeem_whenKeyIsInRedeemScript_shouldReturnKeyIndexPosition() {
+        // Arrange
+        final int EXPECTED_KEY_INDEX = 5;
+        for (Script flyoverRedeemScript : flyoverRedeemScripts) {
+            FlyoverRedeemScriptParser flyoverRedeemScriptParser = new FlyoverRedeemScriptParser(flyoverRedeemScript.getChunks());
+
+            // Act
+            int actualKeyIndex = flyoverRedeemScriptParser.findKeyInRedeem(defaultRedeemScriptKeys.get(EXPECTED_KEY_INDEX));
+
+            // Assert
+            assertEquals(EXPECTED_KEY_INDEX, actualKeyIndex);
+        }
+    }
+
+    @Test
+    public void findKeyInRedeem_whenKeyIsNotInRedeemScript_shouldThrowIllegalStateException() {
+        // Arrange
+        final BtcECKey differentKey = BtcECKey.fromPrivate(BigInteger.valueOf(1000));
+        for(Script flyoverRedeemScript : flyoverRedeemScripts) {
+            FlyoverRedeemScriptParser flyoverRedeemScriptParser = new FlyoverRedeemScriptParser(flyoverRedeemScript.getChunks());
+            try {
+                // Act
+                flyoverRedeemScriptParser.findKeyInRedeem(differentKey);
+            } catch (IllegalStateException actualException) {
+                // Assert
+                assertEquals(IllegalStateException.class, actualException.getClass());
+            }
+        }
```

Comment:
```suggestion
    @Test(expected = IllegalStateException.class)
    public void findKeyInRedeem_whenKeyIsNotInRedeemScript_shouldThrowIllegalStateException() {
        // Arrange
        final BtcECKey differentKey = BtcECKey.fromPrivate(BigInteger.valueOf(1000));
        for(Script flyoverRedeemScript : flyoverRedeemScripts) {
            FlyoverRedeemScriptParser flyoverRedeemScriptParser = new FlyoverRedeemScriptParser(flyoverRedeemScript.getChunks());
            flyoverRedeemScriptParser.findKeyInRedeem(differentKey);
        }
    }
```

Reply:
same for the other tests that expect an exception.

Reply:
At first, I tried it out in this way, however, when the test threw the first exception it broke the loop and didn't keep on with the following redeem script. For this reason, I changed this approach.